src/responses.rs

```diff
@@ -10,6 +10,14 @@ pub fn merge_responses(responses: Vec<Response>) -> Response {
             .add_attributes(response.attributes)
             .add_events(response.events)
             .add_submessages(response.messages);
+        // merge data
+        if let Some(data) = response.data {
+            if merged.data.is_none() {
+                merged.data = Some(data);
+            } else {
+                panic!("Cannot merge multiple responses with data");
```

Comment:
will we get the message from the panic if this fails on chain? @apollo-sturdy

Reply:
Yes because of the "abort" feature in `cosmwasm-std`. But it may be more rusty to return a result. See:
-  https://doc.rust-lang.org/book/ch09-03-to-panic-or-not-to-panic.html#:~:text=The%20panic!,your%20code%20could%20recover%20from.
- https://medium.com/cosmwasm/cosmwasm-1-0-0-finalized-fadc148f9e18
-